step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java

```diff
@@ -331,58 +341,50 @@ static ParameterType of(Type type) {
                     }
                 }
                 if (acceptsList(type)) {
-                    return new ArrayType(of(((ParameterizedType) type).getActualTypeArguments()[0]));
+                    return new ArrayType(Collection.class, of(((ParameterizedType) type).getActualTypeArguments()[0]));
                 }
                 throw new UnsupportedOperationException("do not know how to categorize attributes of type " + type);
             } catch (Exception x) {
-                return new ErrorType(x);
+                return new ErrorType(x, type);
             }
         }
     }
 
     public static final class AtomicType extends ParameterType {
-        private final Class<?> clazz;
         AtomicType(Class<?> clazz) {
-            this.clazz = clazz;
-        }
-        /**
-         * A Java type: either {@link String}, or a {@link Class#isPrimitive} wrapper.
-         */
-        public Class<?> getType() {
```

Comment:
My recommendation was to leave the `getType` method, or some variant, so that a caller would not need to cast `getActualType()` to `Class`.